### PR TITLE
Fixes #14043 - Fixes how product associations are made, to work with multiple organizations correctly.

### PR DESCRIPTION
**Pull request by zjhuntin**
To reproduce:

1. Create two organizations and give them both manifests with overlapping products.

2. On one organization give it a product that has a subscription with many products (EG: I used Red Hat enterprise Linux 7 Server (Kickstart)).

3. In the other organization add products that are in the subscriptions products of the first organizations product. (EG: I enabled Linux Atomic Host RPMs and Oracle java RHEL server).

4. If the change has been applied then when you run hammer activation-key product-content --id [id of first organization's activation key] you should only see products enabled for the first organization. If it's not been applied you should see those and the products enabled in the second organization.

### CODE REVIEW

**Comment by ehelms** [comment]
@jlsherrill @thomasmckay could you double check this logic that it makes sense with respect to our modeling design vs Candlepin?

**Comment by thomasmckay** [comment]
@ehelms - give others a chance to review. commenting implies you need to ack too now

**Comment by ehelms** [comment]
No it doesn't, and I worked on this with Zach, which is why I am requesting
specific folks at a minimum.

**Comment by jlsherrill** [comment]
Yes, this makes perfect sense, we should NEVER look up a product only by cp_id, as its not unique. By Cp_id + org_id should be unique.

**Comment by jlsherrill** [comment]
ACK Merging Thanks @zjhuntin !

**Comment by jlsherrill** [comment]
Actually i'll give @thomasmckay a chance to comment before merging

**Comment by thomasmckay** [comment]
ACK

**Comment by jlsherrill** [comment]
merging thanks @zjhuntin

**Comment by thomasmckay** [comment]
This did not fix the activation key product content. How was it tested?

**Comment by thomasmckay** [comment]
This fix did not fix the UI https://bugzilla.redhat.com/show_bug.cgi?id=1317679

**Comment by thomasmckay** [comment]
It also did not fix hammer. :) I will reopen redmine/bz.

**Comment by ehelms** [comment]
@thomasmckay did you test this with an existing setup or a fresh setup?

**Comment by thomasmckay** [comment]
I tested this on a dev install w/ latest katello/foreman git. UI shows no product content repos, while hammer shows the wrong repos.

**Comment by ehelms** [comment]
Sorry, let me rephrase, did you test this with existing data or start fresh? If you note the change is in the import method which means existing data is already fubared so you either need to reimport all data or start fresh.

**Comment by thomasmckay** [comment]
@ehelms - ACK! I refreshed manifest in org B (w/ breakpoints to confirm code is hit) and it looks good. I'll switch BZs back.
